### PR TITLE
Formatted 1

### PR DESCRIPTION
**Pull request by chick**
### Checklist

- [x] Did you specify the type of improvement?
- [ ] Did you state the API impact?
- [ ] Did you specify the code generation impact?
- [ ] Did you request a desired merge strategy?

### Type of Improvement
Installed .scalafmt.config file
Reformatting all files in src to see what it looks like
Note: Took two passes to stabilize

### Primary Impact
Could break existing PRs

### API Impact
Should be no API changes. 

### Backend Code Generation Impact
No code generation changes

### Desired Merge Strategy
Up in the air

### CODE REVIEW

File: .scalafmt.config [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+danglingParentheses = true
+
+align.tokens = [
+  {
+    code = ":"
+  }
+]
+
+newlines.alwaysBeforeCurlyBraceLambdaParams
 = false
```
Comment (seldridge) [inline .scalafmt.config]: Not sure what this character is in here?

```suggestion
newlines.alwaysBeforeCurlyBraceLambdaParams = false
```

File: .scalafmt.config [hunk 2]
```diff
@@ -0,0 +1,20 @@
+maxColumn = 120
+align = most
+continuationIndent.defnSite = 2
+assumeStandardLibraryStripMargin = true
+docstrings = ScalaDoc
+lineEndings = preserve
+includeCurlyBraceInSelectChains = false
+danglingParentheses = true
+
+align.tokens = [
```
Comment (seldridge) [inline .scalafmt.config]: I think that this is resulting in *only* `:` being aligned. I think the intent was that `:` would *also* be aligned. Concretely, this seems to be causing `=>` to not be aligned which was one of the things I think we wanted?

```suggestion
align.tokens.add = [
```

Comment (chick) [reply .scalafmt.config]: I like it. I will make the change. The weird thing I checked a few => and thought they were aligned.

File: src/main/scala/firrtl/AddDescriptionNodes.scala [hunk 3]
```diff
@@ -31,33 +31,33 @@ private case object EmptyDescription extends Description {
...
-  def mapExpr(f: Expression => Expression): Statement = this.copy(stmt = stmt.mapExpr(f))
-  def mapType(f: Type => Type): Statement = this.copy(stmt = stmt.mapType(f))
-  def mapString(f: String => String): Statement = this.copy(stmt = stmt.mapString(f))
-  def mapInfo(f: Info => Info): Statement = this.copy(stmt = stmt.mapInfo(f))
-  def foreachStmt(f: Statement => Unit): Unit = f(stmt)
-  def foreachExpr(f: Expression => Unit): Unit = stmt.foreachExpr(f)
-  def foreachType(f: Type => Unit): Unit = stmt.foreachType(f)
-  def foreachString(f: String => Unit): Unit = stmt.foreachString(f)
-  def foreachInfo(f: Info => Unit): Unit = stmt.foreachInfo(f)
+  def mapStmt(f:       Statement => Statement):   Statement = f(stmt)
```
Comment (ekiwi) [inline src/main/scala/firrtl/AddDescriptionNodes.scala]: This alignment seems a bit overzealous.

Comment (chick) [reply src/main/scala/firrtl/AddDescriptionNodes.scala]: I think in this case it is a bit, but there are other cases where I think the alignment aids readability.
How overzealous do you think it is.
And would you like to join the formatting meeting at 1:30 on Friday to weigh in?

File: .scalafmt.config (hunk 2 above)
Comment (chick) [reply .scalafmt.config]: The change was made and firrtl had been reformatted again and is pushed

**Comment by chick** [comment]
Will be revisited when PR queue diminishes

**Pull request closed without merging**